### ISSUE TITLE
Transformation error (Unexpected token) and wrong behaviours

### ISSUE THREAD

**Issue by DarthGigi** [getsentry/sentry-migr8#52]
### Environment

Svelte/Kit

### Steps to Reproduce

1. Run `pnpx @sentry/migr8@latest`
2. Select `Apply all transformations`

### Expected Result

The migrate command changes the files that are actually necessary to be changed and in the right way

### Actual Result

For each component I have in my `routes` folder, I get the following error:

```ts
ERR /src/routes/bug-report.svelte Transformation error (Unexpected token (34:1))
SyntaxError: Unexpected token (34:1)
```

This is because for some reason `</string>` is being added just before `</script>` when transforming the files:
![Screenshot 2024-05-16 at 9  43 31@2x](https://github.com/getsentry/sentry-migr8/assets/47110839/12cfc6f3-db5c-406a-9824-b2898c82012a)

This doesn't happen in sveltekit's route files (+layout and +page.svelte) files, however, it does in `+error.svelte` files.

Another thing is that for some reason it tries to "fix" self-closing elements.
If we have `<div />` for example, it changes it to `<div></div>`, while it is true that `<div />` is [not valid HTML](https://github.com/sveltejs/svelte/issues/11052), this should not apply to components:

![Screenshot 2024-05-16 at 9  51 53@2x](https://github.com/getsentry/sentry-migr8/assets/47110839/fbd7c8fb-591e-44c8-bcd1-9a051fc96d8c)

Why is Sentry changing my markup? Isn't it supposed to just change it's own stuff and only in `<script>` tags?

Another thing you can see in the screenshot is that Sentry added `</svelte:window>` to the bottom of the file:
1. `<svelte:window/>` does not accept children, so it must be always closed/self closed without anything as it's child
2. `<svelte:window/>` is already closed, why does Sentry think it's not?

This migrate command changed about 170 files (mostly .svelte files since I'm using shadcn svelte components) which makes it not worth going over the stuff that Sentry changed to actually migrate from v7 to v8 and to check if there are any todo's left.

Feel free to change the title of this issue to a more fitting one.

**Comment by Lms24** [comment]
Hey @DarthGigi thanks for writing in and apologies for the late reply! 

> Why is Sentry changing my markup? Isn't it supposed to just change it's own stuff and only in <script> tags?

Definitely not on purpose. Ideally, we change as little as possible. My best guess is that our parser has problems with svelte files and I'm actually considering excluding svelte files from the transformations for the moment. At least in basic svelte apps there shouldn't be too much need to use Sentry code in svelte components.

> This is because for some reason </string> is being added just before </script> when transforming the files

Hmm that's also a weird one. I'll check what's going wrong here.

**Comment by Florinstruct** [comment]
Same here. Migration script deletes or rewrites random seeming sections of our codebase. I tried to figure out which option ("transformer") is responsible, but more than one seems to trigger this.

Even if you only `Add migration comments`, the script will rewrite and break over 400 files and break 150 of them.

![image](https://github.com/getsentry/sentry-migr8/assets/113526045/aebd48c0-6aed-426a-bbae-b4ad097d5747)

**Comment by DarthGigi** [comment]
Any updates on this issue?

**Comment by Lms24** [comment]
Hi, again sorry for the late reply. Our GH issue monitoring tool wasn't aware of this repo and hence replies slipped through. I believe this was fixed in v0.2.4 of migr8 (via #69 and #71). Closing this issue as completed for the time being. Please feel free to tag me if this needs to be re-opened. Thanks!
